src/main/java/org/opensearch/knn/index/codec/nativeindex/QuantizationIndexUtils.java

```diff
@@ -33,7 +33,11 @@ static Object processAndReturnVector(KNNVectorValues<?> knnVectorValues, IndexBu
                 knnVectorValues.getVector(),
                 indexBuildSetup.getQuantizationOutput()
             );
-            return indexBuildSetup.getQuantizationOutput().getQuantizedVector();
+            /**
+             * Returns a copy of the quantized vector. This is because of during transfer same vectors was getting
+             * added due to reference.
+             */
+            return indexBuildSetup.getQuantizationOutput().getQuantizedVectorCopy();
```

Comment:
Can we update the unit test for this class to make sure we always return a copy? we need a `assertNotSame` check in the unit test of this class